src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -397,28 +388,17 @@ public static void CreateAssociation(object primary, object secondary)
                 throw new ArgumentException(SR.TypeDescriptorSameAssociation);
             }
 
-            if (s_associationTable == null)
-            {
-                lock (s_internalSyncObject)
-                {
-                    if (s_associationTable == null)
-                    {
-                        s_associationTable = new WeakHashtable();
-                    }
-                }
-            }
-
-            IList associations = (IList)s_associationTable[primary];
+            IList associations = (IList)s_associationTableLazy.Value[primary];
 
             if (associations == null)
             {
-                lock (s_associationTable)
+                lock (s_associationTableLazy)
```

Comment:
> On the lines of s_defaultProviders reversal discussion above, does s_associationTable really require lazy initialisation. The initialisation is just object creation and does not do any heavy code that may cause delays. So how about making it readonly and initialize in-place?

I would like to avoid changing lazy/prompt semantics in this change.  The original developer likely had a reason for what they did (lazy vs non-lazy), and so as part of this change I would like to keep it that way.  If subsequently you'd like to investigate with appropriate performance tests whether it would be worth changing, that'd be fine as a separate, isolated thing to explore.